File: lib/data/models/food_search_model.dart
import 'package:calory_tool/data/models/food_model.dart';

class FoodSearchModel {
  const FoodSearchModel({required this.pageNumber, required this.foods});

  factory FoodSearchModel.fromJson(Map<String, dynamic> json) {
    final foodsJson = json['food'] as List?;
    final foods =
        foodsJson
            ?.map((e) => FoodModel.fromJson(e as Map<String, dynamic>))
            .toList() ??
        [];
    return FoodSearchModel(
      pageNumber: json['pageNumber'] as int?,
      foods: foods,
    );
  }

  final int? pageNumber;
  final List<FoodModel> foods;
}

File: server/src/models/foods/FoodSearchModel.js
import FoodModel from "./FoodModel.js";

class FoodSearchModel {
  constructor({ pageNumber, foods }) {
    this.pageNumber = pageNumber;
    this.foods = foods;
  }

  static fromJson(json) {
    const foodsSearchJson = json["foods_search"];
    const foodsJson = foodsSearchJson?.["results"] ?? null;
    const foodJson = foodsJson?.["food"] ?? null;
    return new FoodSearchModel({
      pageNumber: foodsJson?.["page_number"] ? parseInt(foodsJson["page_number"]) : null,
      food: foodJson?.map(food => FoodModel.fromJson(food)) ?? null,
    });
  }

  toJson() {
    return {
      page_number: this.pageNumber,
      foods: this.foods ? this.foods.map(food => food.toJson()) : null,
    };
  }
}

export default FoodSearchModel;
